backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/controller/SpecificationController.kt

```diff
@@ -56,9 +76,42 @@ class SpecificationController @Autowired constructor(
         return specificationIdFromPath
     }
 
+    @PutMapping("/{specificationId}/chmod/{userId}")
+    fun chmodVersion(@PathVariable specificationId: String,
+                     @PathVariable userId: String,
+                     @RequestParam(value = "grantRead", defaultValue = "false") grantRead: String,
+                     @RequestParam(value = "grantWrite", defaultValue = "false") grantWrite: String
+    ) {
+        logger.info(grantRead)
+        logger.info(grantWrite)
+        logger.info(specificationId)
+        logger.info(userId)
+
+        try {
+            val idAsLong = specificationId.toLong()
+            // TODO: Add exception if the target principal or specification doesn't exist
+            val targetUserSid = PrincipalSid(userId)
+            changePermission(idAsLong, grantRead.toBoolean(), targetUserSid, BasePermission.READ)
+            changePermission(idAsLong, grantWrite.toBoolean(), targetUserSid, BasePermission.WRITE)
+
+        } catch (exc: NumberFormatException) {
+            throw IdNotCastableToLongException()
+        }
+
+    }
+
+    private fun changePermission(specificationId: Long, grant: Boolean, sid: PrincipalSid, permission: Permission) {
+        if (grant)
+            permissionManager.addPermission(specificationId, sid, permission)
```

Comment:
Important rule: Always use braces! It's more explicit and thus easier to read. And it's harder to make an error...